data/src/main/java/io/gnosis/data/models/Transaction.kt

```diff
@@ -19,26 +29,47 @@ sealed class Transaction {
 
     data class SettingsChange(
         override val status: TransactionStatus,
+        override val confirmations: Int?,
         val dataDecoded: DataDecodedDto,
         val date: String?,
         val nonce: BigInteger
-    ) : Transaction()
+    ) : Transaction() {
+        fun isChangeMasterCopy(): Boolean {
+            return "changeMasterCopy" == dataDecoded.method
+        }
+
+        fun isSetFallBackHandler(): Boolean {
+            return "setFallbackHandler" == dataDecoded.method
+        }
+    }
 
     data class Transfer(
         override val status: TransactionStatus,
+        override val confirmations: Int?,
         val recipient: Solidity.Address,
         val sender: Solidity.Address,
         val value: BigInteger,
         val date: String?,
-        val tokenInfo: ServiceTokenInfo
+        val tokenInfo: ServiceTokenInfo?,
+        val nonce: BigInteger?
     ) : Transaction()
+
+    fun isCompleted(): Boolean =
+        when (status) {
+            AwaitingConfirmations,
+            AwaitingExecution,
+            Pending -> false
+            Success,
+            Failed,
+            Cancelled -> true
+        }
 }
 
-enum class TransactionStatus {
-    AwaitingConfirmation,
-    AwaitingExecution,
-    Cancelled,
-    Failed,
-    Success,
-    Pending // Not supported yet as these correspond to the ones issued from the device
+enum class TransactionStatus(val displayString: String) {
+    AwaitingConfirmations("Awaiting confirmations"),
```

Comment:
better not to mix models and ui related elements like display strings. We should be using localized string resources and the decision which string to show should be done at the view model level